src/common/devcard.ts

```diff
@@ -65,12 +66,12 @@ export async function getDevCardData(
   userId: string,
   con: Connection,
 ): Promise<DevCardData> {
-  const [user, articlesRead, tags, sourcesLogos, rank] = await Promise.all([
-    fetchUser(userId),
+  const user = await con.getRepository(User).findOne(userId);
```

Comment:
If `user` is not defined, I think this function should fail

Reply:
Just checking, but this function can never be called if not authorized right?
Because all the sub functions also use the user.

I can add a hard return fail though

Reply:
You are right. You can use `findOneOrFail`, this function should do it for you. Might named a bit different